### DISCUSSION TITLE
Any guide on implementing Okta/Auth0 or similar services?

### DISCUSSION THREAD

**Discussion by theMadness** [api-platform/api-platform#2048] [Q&A]
I'm having some challenges getting one such authentication method implemented, particularly when it comes to it integrating with next.js and ssr pages. Anyone had some success with it and is willing to share some insights?

**Comment by ngrafa** [comment]
> I'm having some challenges getting one such authentication method implemented, particularly when it comes to it integrating with next.js and ssr pages. Anyone had some success with it and is willing to share some insights?

We are using Okta but still in development, so I can't show any live examples but I can share the approach we have taken. This is very top level. I could share some code examples, but I need to get permission first.

In our early example using Vue 3 for the frontend, we go through the Okta flow, which redirects to the Okta sign-in page. Once the user completes the log in, the callback function takes the user back to a route on the Vue 3 app, which stores some basic info from the Okta claim into the state management (vuex) if authentication is successful. Once auth is successful, the guarded routes become accessible, but any API calls from Vue to API Platform need to have the JWT Bearer token sent to the endpoint on the Symfony/API Platfom app. 

On the Symfony side, we have a custom authenticator, such as `App\Security\TokenAuthenticator.php` which creates the `SelfValidatingPassport` object once the JWT Bearer token has been verified using the package `okta/jwt-verifier`. We have a separate Service to do all the Okta Verifier heavy lifting.

It was very tricky as we started this before Symfony 5, and auth has been evolving in Symfony 5 and 6, but also our use-case is complex as our system will need to support multiple identity providers, including some owned by our customers, but hopefully the information helps.
